pkg/agent/controller/egress/egress_controller.go

```diff
@@ -848,9 +875,13 @@ func (c *EgressController) syncEgress(egressName string) error {
 		// Ensure the Egress IP is assigned to the system. Force advertising the IP if it was previously assigned to
 		// another Node in the Egress API. This could force refreshing other peers' neighbor cache when the Egress IP is
 		// obtained by this Node and another Node at the same time in some situations, e.g. split brain.
-		if err := c.ipAssigner.AssignIP(desiredEgressIP, egress.Status.EgressNode != c.nodeName); err != nil {
+		exists, err := c.ipAssigner.AssignIP(desiredEgressIP, egress.Status.EgressNode != c.nodeName)
+		if err != nil {
 			return err
 		}
+		if exists && c.record != nil {
+			c.record.Eventf(egress, corev1.EventTypeNormal, "IPAssigned", "Assigned egress %s with IP %s on node %s", egress.Name, desiredEgressIP, desiredNode)
+		}
 	} else {
 		// Unassign the Egress IP from the local Node if it was assigned by the agent.
 		if err := c.ipAssigner.UnassignIP(desiredEgressIP); err != nil {
```

Comment:
I think we can send the IPUnassigned event here as it's more accurate. Then no need to send the IPUnassigned event in other places (though it don't cover the node crash case, but it makes more sense in a way as in that case the IP was not really unassigned from the node)
Similarly, it should only send the event when it actually unassigned the IP.